src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSource.cs

```diff
@@ -111,6 +115,10 @@ public class HttpSource : IDisposable
                             log.LogWarning(message);
                         }
                     }
+                    else
+                    {
+                        ProtocolDiagnostics.RaiseHttpMissCacheEvent(cacheResult.MaxAge == TimeSpan.Zero, isExpired);
```

Comment:
Here is `cacheResult.MaxAge == TimeSpan.Zero` means it wants to bypass the cache.